### PR TITLE
Fix projection pushdown when unsupported column exists in BigQuery

### LINKED ISSUE #15093: Fix projection pushdown when unsupported column exists in BigQuery
Projection pushdown happens when all columns are selected except for unsupported types.
```sql
-- Prepare a table in BigQuery - json type is unsupported in the connector
CREATE TABLE test.test_unsupported (c1 int64, c2 int64, c3 json);
INSERT INTO test.test_unsupported VALUES (1, 2, json '{}');
```

```sql
-- The connector retrieves only c1 column from BigQuery
SELECT c1 FROM bigquery.test.test_unsupported;

-- The connector retrieves all columns including c3 column from BigQuery
SELECT c1, c2 FROM bigquery.test.test_unsupported;
```

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #15093

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
I manually confirmed the internal projection because it's a little hard to verify in tests.

**Comment by ebyhr** [comment]
@Praveen2112 @hashhar @vlad-lyutenko Gentle reminder.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java [hunk 1]
```diff
@@ -230,12 +232,20 @@ public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTable
...
+        }
         return new BigQueryTableHandle(new BigQueryNamedRelationHandle(
                 schemaTableName,
                 new RemoteTableName(tableInfo.get().getTableId()),
                 tableInfo.get().getDefinition().getType().toString(),
-                getPartitionType(tableInfo.get().getDefinition()),
-                Optional.ofNullable(tableInfo.get().getDescription())));
+                partitionType,
+                Optional.ofNullable(tableInfo.get().getDescription())))
+                .withProjectedColumns(columns.build());
```
[the remark this answers is not in the document]
Comment (vlad-lyutenko) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java:248]: Ok I see now: 
```
    public static Optional<Set<Symbol>> pruneInputs(Collection<Symbol> availableInputs, Collection<Expression> expressions)
    {
        Set<Symbol> availableInputsSet = ImmutableSet.copyOf(availableInputs);
        Set<Symbol> prunedInputs = Sets.filter(availableInputsSet, SymbolsExtractor.extractUnique(expressions)::contains);

        if (prunedInputs.size() == availableInputsSet.size()) {
            return Optional.empty();
        }

        return Optional.of(prunedInputs);
    }
```
size input == size output and we didn't apply projection

**Comment by ebyhr** [comment]
@Praveen2112 @hashhar Gentle reminder.

**Review by hashhar — APPROVED** [review]

**Pull request merged**